hyperstart/api/grpc/hyperstart.proto

```diff
@@ -5,15 +5,17 @@ package grpc;
 import "google/protobuf/empty.proto";
 
 // To generate hyperstart.pb.go, run:
-// $ protoc --go_out=plugins=grpc:. hyperstart.proto
+// $ go get -u github.com/google/protobuf
+// $ protoc -I=$GOPATH/src/github.com/google/protobuf/src/ -I. --go_out=plugins=grpc:. hyperstart.proto
 
 // unstable
 service HyperstartService {
 	// execution
-	rpc AddContainer(AddContainerRequest) returns (google.protobuf.Empty);
-	rpc AddProcess(AddProcessRequest) returns (google.protobuf.Empty);
-	rpc SignalProcess(SignalProcessRequest) returns (google.protobuf.Empty);
-	rpc WaitProcess(WaitProcessRequest) returns (WaitProcessResponse); // wait & reap like waitpid(2)
+	rpc CreateContainer(CreateContainerRequest) returns (google.protobuf.Empty);
+	rpc StartContainer(StartContainerRequest) returns (google.protobuf.Empty);
+	rpc Exec(ExecRequest) returns (google.protobuf.Empty);
```

Comment:
We should be able to combine StartContainer and Exec just like above AddProcess does. Is there any special requirement to separate starting container and exec?

Reply:
Yes we can merge both `StartContainer()` and `Exec()` into `StartProcess()` since that's what this is. We have `CreateContainer()` which really means that we are creating a container but without starting any process in it. And we could have `StartProcess()` having a flag to make the difference between the container process or any other process that we want to start.

But at the same time, I think it might be okay to have separate functions for each case since there are 2 distincts use cases. The question is, do we mind having some kind of overlapping behaviors ? I think it's okay, the agent implementation will rely on the same code for the main part of those 2 functions.

Reply:
@bergwolf From a code perspective, they will look very similar, but from an API perspective they have different semantics (Their arguments are even different, you don't need to specify a Process when starting an already created container). So I think the API is clearer and more explicit with those 2 separated.

Reply:
I tend to think we should keep the API list short. But I can live with this. @laijs @gnawux @gao-feng what do you guys think?